.github/workflows/ci.yml

```diff
@@ -1,34 +1,46 @@
 name: CI
 
-on:
-  push:
-    branches: master
-  pull_request:
-    branches: master
+on: [push, pull_request]
 
 jobs:
-  checks:
-    name: Documentation build
-    runs-on: ubuntu-latest
-    steps:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: ["windows-latest", "ubuntu-latest", "macos-latest"]
+        python-version: ["3.6", "3.7", "3.8"]
+
+    env:
+      PYTHON_VERSION: ${{ matrix.python-version }}
+      PARALLEL: "true"
+      COVERAGE: "false"
 
-    - name: Checkout
-      uses: actions/checkout@v2
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
 
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.7
+      - name: Setup Conda Environment
+        uses: goanpeca/setup-miniconda@v1.7.0
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ matrix.python-version }}
+          environment-file: continuous_integration/environment-${{ matrix.python-version }}.yaml
+          activate-environment: test-environment
+          auto-activate-base: false
 
-    - name: Install Dask
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install -e .
+      - name: Install
+        shell: bash -l {0}
+        run: source continuous_integration/scripts/install.sh
 
-    - name: Install doc dependencies
-      run: python -m pip install -r docs/requirements-docs.txt
+      - name: Run tests
+        shell: bash -l {0}
+        run: source continuous_integration/scripts/run_tests.sh
 
-    - name: Build docs
-      run: |
-        cd docs
-        make html
+      - name: Coverage
+        shell: bash -l {0}
+        if: ${{ matrix.os == 'ubuntu-latest' &&  matrix.python-version == '3.8' }}
+        env:
+          COVERAGE: "true"
+          PARALLEL: "false"
+        run: source continuous_integration/scripts/coverage.sh
```

Comment:
Currently we're getting a `No data to report.` error when this runs. We might consider switching over to using [Coverall's GitHub Action](https://github.com/marketplace/actions/coveralls-github-action). Additionally, while coverage is nice to have, I'd be okay commenting this out for now and addressing it in a follow-up PR if it means we can merge this sooner